content/articles/building-web-application-with-kotlin-and-typescript/index.md

```diff
@@ -0,0 +1,191 @@
+
+It's necessary to outline the Gradle metadata so they can be utilized to fit the desired dependencies. Grаdle metаdаta is commonly used in Kоtlin Multiрlаtfоrm builds to treat Grаdle dependants with the appropriate аrtifасts.
+
+```
+//from file: client/build.gradle.kts
+val nodeKotlin by configurations.creating {
+  attributes {
+    attribute(KotlinPlatformType.attribute, KotlinPlatformType.js)
+    attribute(
+      usage.USAGE_ATTRIBUTE,
+      mission.objects.named(
+          usage::class.java,
+          KotlinUsages.KOTLIN_RUNTIME
+      )
+    )
+  }
+}
+```
+
+The configuration above is for coordination of the JS-built platform artifacts used at runtime. The KOTLIN_RUNTIME utilization is for the Gradle to gather all dependencies transitively required at runtime.
+
+Dependencies are added to the required Kotlin modules as per the configuration:
+
+```
+//from file: client/build.gradle.kts
+dependencies {
+  nodeKotlin(project(":information"))
+}
+```
+The nodeKotlin dependencies require to be resolved and artifacts that contain the JavaScript codes should be unpacked to the node_modules directory.
+
+### Unpacking the Kotlin modules
+
+Multiplatform Gradle plugin provides already integrated tasks that helps us to run both the Node and Yarn commands. Although the Kotlin plugin makes the node_modules directory containing several javascript modules it's hard to make the angular code use them since it requires its node_module closer to both its angular.json and the package.json files.
+
+Creating hard links remains an option fooling the angular to think node_modules are where it wants them to be. The following are tasks that create the required node_modules directory, placing them in the Kotlin-JS module.
+
+- yarnInstall: A call to yarn installs the expected JS modules.
+
+- UnpackKotlinJs: JS files are unpacked from nodekotlin dependencies.
+
+- nodebuild: Angular/node.js build gets invoked.
+
+### Using Kotlin generated modules in TypeScript code
+
+Now that the Kotlin-JS modules are in the node_modules directory we can create the Angular/TypeScript code. There are a few other ways to do it but we will consider using the import statement method since the javascript module does most of the work.
+
+Following is a way of importing generated Kotlin modules to a TypeScript file:
+
+```
+import * as info_js from 'example.addressbook-information';
+import info = info_js.example.addressbook.information;
+...
+let contact = new info.contact('alias')
+```
+import statement on the second time is not much important it also develops a good alias for the content of the modules which helps to reduce more writing of fully qualified names all the time we use Kotlin generated modules.

```

Comment:
refrase to "Import statement on the second time is not much important. It also develops a good alias for the content of the modules, which helps to reduce more writing of fully qualified names all the time we use Kotlin generated modules."